src/libcore/iter/iterator.rs

```diff
@@ -1506,12 +1563,10 @@ pub trait Iterator {
         Self: Sized,
         F: FnMut(Self::Item) -> bool
     {
-        for x in self {
-            if f(x) {
-                return true;
-            }
-        }
-        false
+        self.try_fold((), move |(), x| {
+            if f(x) { SearchResult::Found(()) }
+            else { SearchResult::NotFound(()) }
+        }).is_found()
```

Comment:
Just curious why all and any are different (one uses SearchResult and the other not)

Reply:
I think it's because my brain thinks in the `true` case, so `all` is `continue` on "success" (Result) but `any` is `break` on "success" (SearchResult).  That is, of course, a super-fuzzy statement, since the opposites also work just as well.